README.md

```diff
@@ -28,23 +29,109 @@ The package could be installed with [Composer](https://getcomposer.org):
+
+Minimal configuration example:
+
+```php
+return [
+    'yiisoft/queue' => [
+        'handlers' => [
+            'handler-name' => [FooHandler::class, 'handle'],
+        ],
+    ],
+];
+```
+
+[Advanced configuration with yiisoft/config](docs/guide/en/configuration-with-config.md)
+
+#### Manual configuration
+
+For setting up all classes manually, see the [Manual configuration](docs/guide/en/configuration-manual.md) guide.
+
+### 3. Prepare a handler
+
+You need to create a handler class that will process the queue messages. The most simple way is to implement the `HandleInterface`. Let's create an example for remote file processing:
+
+```php
+use Yiisoft\Queue\Handler\HandleInterface;
+use Yiisoft\Queue\Message\MessageInterface;
+
+final readonly class RemoteFileHandler implements HandleInterface
+{
+    private string $absolutePath;
+
+    // These dependencies will be resolved on handler creation by the DI container
+    public function __construct(
+        private FileDownloader $downloader,
+        private FileProcessor $processor,
+    ) {}
+
+    // Every received message will be processed by this method
+    public function handle(MessageInterface $downloadMessage): void
+    {
+        $fileName = $downloadMessage->getData()['destinationFile'];
+        $localPath = $this->downloader->download($fileName);
+        $this->processor->process($localPath);
+    }
+}
+```
+
+### 4. Send (produce/push) a message to a queue
+
+To send a message to the queue, you need to get the queue instance and call the `push()` method. Typically with Yii Framework you'll get a `Queue` instance as a dependency of a service.
+
+```php
+
+final readonly class Foo {
+    public function __construct(private QueueInterface $queue) {}
+    
+    public function bar(): void
+    {
+        $this->queue->push(new Message(
+            // The first parameter is the handler name that will process this concrete message
+            RemoteFileHandler::class,
+            // The second parameter is the data that will be passed to the handler.
+            // It should be serializable to JSON format
+            ['destinationFile' => 'https://example.com/file-path.csv'],
+        ));
+    }
+}
+```
+
+### 5. Handle queue messages
+
+By default Yii Framework uses [yiisoft/yii-console](https://github.com/yiisoft/yii-console) to run CLI commands. If you installed [yiisoft/app](https://github.com/yiisoft/app) or [yiisoft/app-api](https://github.com/yiisoft/app-api), you can run the queue worker with on of these two commands:
+
+```bash
+./yii queue:run # Handle all existing messages in the queue
+./yii queue:listen # Start a daemon listening for new messages permanently
+```
+
+> In case you're using the SynchronosAdapter for development purposes, you should not use these commands, as you have no asynchronous processing available. The messages are processed immediately when pushed.
```

Comment:
```suggestion
> In case you're using the `SynchronosAdapter` for development purposes, you should not use these commands, as you have no asynchronous processing available. The messages are processed immediately when pushed.
```